### PR TITLE
Use newer version of io-sim

### PR DESCRIPTION
**Pull request by coot**
# Description

Update to newer version of `io-sim` (see input-output-hk/io-sim#24).

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by coot** [comment]
Requires both:
* input-output-hk/io-sim#24
* input-output-hk/typed-protocols#10

**Review by bolt12 — APPROVED** [review]

File: ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/NormalForm.hs [hunk 1]
```diff
@@ -14,14 +15,21 @@ module Ouroboros.Consensus.Util.MonadSTM.NormalForm (
...
 import           Ouroboros.Consensus.Util.MonadSTM.StrictMVar hiding
                      (newEmptyMVar, newEmptyMVarWithInvariant, newMVar,
                      newMVarWithInvariant)
 
-import qualified Control.Monad.Class.MonadSTM.Strict as Strict
+import           Control.Concurrent.Class.MonadSTM.Strict.TMVar as StrictSTM hiding
+                     (newTMVar, newTMVarIO)
+import           Control.Concurrent.Class.MonadSTM.Strict.TVar as StrictSTM hiding
+                     (newTVar, newTVarIO, newTVarWithInvariantIO)
+-- TODO: use strict versions of 'TQueue' and 'TBQueue'
```
Comment (dnadales) [inline ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/NormalForm.hs:27]: Could we give a rationale on why this is required? Eg what would happen if we don't use the strict versions of these queue types?

File: ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/StrictMVar.hs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 {-# LANGUAGE TupleSections     #-}
 {-# LANGUAGE TypeFamilies      #-}
 
+-- TODO: this module ought to use 'MonadMVar', and be moved to `strict-stm`.
```
Comment (dnadales) [inline ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/StrictMVar.hs:8]: It'd help whoever needs to do this to provide a rationale for this.

**Review by dnadales — APPROVED** [review]
LGTM modulo adding some more context in the two TODO's we have introduced.

File: ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/StrictMVar.hs (hunk 2 above)
Comment (coot) [reply ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/StrictMVar.hs]: I updated the todo.

File: ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/NormalForm.hs (hunk 1 above)
Comment (coot) [reply ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/NormalForm.hs]: I updated the todo.

**Pull request merged**